cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -137,6 +152,8 @@ void Run(const std::vector<string>& bag_filenames) {
     rosbag::View view(bag);
     const ::ros::Time begin_time = view.getBeginTime();
     const double duration_in_seconds = (view.getEndTime() - begin_time).toSec();
+    // The message processing loop publishes the clock.
```

Comment:
> publishing during FinishTrajectory(), SerializeState() and waiting for shutdown

I wanted to keep it this way because I found it more readable to describe as "republish clock always, except during message processing" rather than "publish during finish trajectory finalization, state serialization and waiting for shutdown (and eventually during something else we might add at the end)". I initially mentioned to you only trajectory finalization as being problematic, but what I had in mind was more along these lines. However, I can add it before SerializeState(), if you think it's more readable, no problem. :)